OpenRA.Mods.Common/Widgets/Logic/Ingame/IngameMenuLogic.cs

```diff
@@ -144,8 +144,9 @@ public IngameMenuLogic(Widget widget, World world, Action onExit, WorldRenderer
 				closeMenu();
 			};
 			var surrenderButton = menu.Get<ButtonWidget>("SURRENDER");
+			var surrenderVisCheck = world.LocalPlayer == null || world.LocalPlayer.WinState != WinState.Undefined || world.Map.Visibility == MapVisibility.MissionSelector;
 			surrenderButton.IsVisible = () => world.Type == WorldType.Regular;
-			surrenderButton.IsDisabled = () => (world.LocalPlayer == null || world.LocalPlayer.WinState != WinState.Undefined) || hasError;
+			surrenderButton.IsDisabled = () => surrenderVisCheck || hasError;
```

Comment:
Moving this code out into `surrenderVisCheck` means it will be evaluated once when the menu opens but not after that. Therefore if you win/lose whilst the menu is open the button won't become disabled now. Is that intentional?